src/modules/formBuilder/components/itemEditor/FormEditorItemProperties.tsx

```diff
@@ -0,0 +1,380 @@
+  onSave: (item: FormItem, initialLinkId: string) => void;
+  errorState?: ErrorState;
+  onDiscard: () => void;
+  // React Hook Form handlers
+  handlers: UseFormReturn<FormItemState, any>;
+}
+
+const FormEditorItemProperties: React.FC<FormEditorItemPropertiesProps> = ({
+  initialItem,
+  definition,
+  saveLoading,
+  errorState,
+  onSave,
+  onDiscard,
+  handlers,
+}) => {
+  // React Hook Form handlers (useForm return object)
+  const {
+    control,
+    getValues,
+    handleSubmit,
+    watch,
+    formState: { isDirty },
+  } = handlers;
+
+  // Item type is not allowed to change in this form
+  const [itemTypeValue] = useState<ItemType>(initialItem.type);
+
+  // Monitor changes to the FormItem.component field
+  const itemComponentValue = watch('component');
+
+  const isAssessment = useMemo(
+    () => (Object.values(AssessmentRole) as [string]).includes(definition.role),
+    [definition.role]
+  );
+
+  const componentOverridePicklist = useMemo(() => {
+    return validComponentsForType(itemTypeValue).map((component) => ({
+      code: component,
+      label: startCase(component.toLowerCase()),
+    }));
+  }, [itemTypeValue]);
+
+  return (
+    <form
+      // handleSubmit validates input before calling onSubmit
+      onSubmit={handleSubmit((item) =>
+        // FIXME: we cast to FormItem because of DeepPartial difference. is this safe?
+        onSave(item as FormItem, initialItem.linkId)
+      )}
+    >
+      <Box sx={{ p: 2 }}>
+        <Stack gap={2}>
+          {itemTypeValue !== ItemType.Group && (
+            <FormEditorItemPreview control={control} />
+          )}
+          <Divider />
+        </Stack>
+
+        <Stack gap={2} sx={{ maxWidth: MAX_INPUT_AND_LABEL_WIDTH, mt: 2 }}>
+          {/* TODO standardize heading typography, use hX */}
+          <Typography>
+            Properties
+            {import.meta.env.MODE === 'development' && (
+              // Temp debug button to log form state
+              <Button
+                size='small'
+                variant='outlined'
+                sx={{ float: 'right', typography: 'body2' }}
+                // eslint-disable-next-line no-console
+                onClick={() => console.log(getValues())}
+              >
+                debug
+              </Button>
+            )}
+          </Typography>
+          {componentOverridePicklist.length > 0 && (
+            <Controller
+              name='component'
+              control={control}
+              // rules={{ required: 'need this' }}
+              render={({ field: { ref, ...field }, fieldState: { error } }) => (
+                <SelectOption
+                  label='Component Override'
+                  options={componentOverridePicklist}
+                  {...field}
+                  textInputProps={{
+                    helperText: error?.message,
+                    error: !!error,
+                    inputRef: ref,
+                  }}
+                />
```

Comment:
There's a fair amount of repeated code here. I tried to pull out the common parts (passing ref and errors to `textInputProps`) but it was finicky to get the function types right. I can try again though. Not sure if this is a common issue with RHF maybe there are patterns.